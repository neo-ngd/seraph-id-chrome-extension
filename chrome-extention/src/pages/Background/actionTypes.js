export const CREATE_CLAIM_ALIAS = 'CREATE_CLAIM_ALIAS';
export const ASK_CLAIM_ALIAS = 'ASK_CLAIM_ALIAS';
export const SET_PASSWORD_ALIAS = 'SET_PASSWORD_ALIAS';
export const CHECK_PASSWORD_ALIAS = 'CHECK_PASSWORD_ALIAS';
export const GET_PASSWORD_ALIAS = 'GET_PASSWORD_ALIAS';

export const SET_WALLET = 'SET_WALLET';
export const SET_PASSWORD = 'SET_PASSWORD';
export const SET_CLAIM = 'SET_CLAIM';
export const DESTROY_CLAIM = 'DESTROY_CLAIM';
export const TOGGLE_DIALOG = 'TOGGLE_DIALOG';
export const SEND_ERROR = 'SEND_ERROR';
export const DESTROY_ERROR = 'DESTROY_ERROR';
export const SET_HASH = 'SET_HASH';
export const DESTROY_HASH = 'DESTROY_SESSION';
export const SET_SESSION = 'SET_SESSION';
export const DESTROY_SESSION = 'SET_SESSION';
