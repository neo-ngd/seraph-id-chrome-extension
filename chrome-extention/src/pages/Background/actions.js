import {
  ASK_CLAIM_ALIAS,
  CREATE_CLAIM_ALIAS,
  SET_PASSWORD,
  SET_WALLET,
  SET_CLAIM,
  TOGGLE_DIALOG,
  SEND_ERROR,
  DESTROY_ERROR,
  DESTROY_CLAIM,
  SET_PASSWORD_ALIAS,
  CHECK_PASSWORD_ALIAS, SET_HASH, DESTROY_HASH, GET_PASSWORD_ALIAS, SET_SESSION, DESTROY_SESSION
} from './actionTypes';

export const getEncryptedPassword = () =>
    ({ type: GET_PASSWORD_ALIAS });

export const checkPassword = password =>
    ({ type: CHECK_PASSWORD_ALIAS, password });

export const setEncryptedPassword = password =>
    ({type: SET_PASSWORD_ALIAS, password});

export const setPassword = password =>
  ({type: SET_PASSWORD, password});

export const setExportedWallet = exportedWalletJSON =>
  ({type: SET_WALLET, exportedWalletJSON });

export const createClaim = (data, schemaName) =>
  ({type: CREATE_CLAIM_ALIAS, data, schemaName });

export const askClaim = ({schemaName, issuerDID, verifierName}) =>
  ({type: ASK_CLAIM_ALIAS, schemaName, issuerDID, verifierName});

export const setClaim = claim =>
  ({type: SET_CLAIM, claim});

export const destroyClaim = () =>
    ({ type: DESTROY_CLAIM });

export const toggleDialog = dialog =>
  ({ type: TOGGLE_DIALOG, dialog });

export const sendError = error =>
    ({ type: SEND_ERROR, error});

export const destroyError = () =>
    ({ type: DESTROY_ERROR });

export const setHash = hash =>
    ({ type: SET_HASH, hash});

export const destroyHash = () =>
    ({ type: DESTROY_HASH });

export const setSession = session =>
    ({ type: SET_SESSION, session});

export const destroySession = session =>
    ({ type: DESTROY_SESSION, session});


