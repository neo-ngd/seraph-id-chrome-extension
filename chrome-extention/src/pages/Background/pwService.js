// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

/**
 * The password service.
 * It stores and shares the password.
 */
class PasswordService {
    constructor() {
        this.pw = null;
    }

    set password(pw) {
        this.pw = pw;
    }

    get password() {
        return this.pw;
    }
}

export const pwService = new PasswordService();
