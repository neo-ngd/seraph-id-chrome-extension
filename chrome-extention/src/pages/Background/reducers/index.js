import '../../../assets/icons/icon128';

import { combineReducers } from 'redux';

import { wallet, password, claim, dialog } from './reducers';

export default combineReducers({
  wallet,
  password,
  claim,
  dialog
});
