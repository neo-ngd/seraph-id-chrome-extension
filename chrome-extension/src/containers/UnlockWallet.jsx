// Copyright (c) 2019 Swisscom Blockchain AG
// Licensed under MIT License

import React, { useEffect, useState } from 'react';
import { Box, Link, makeStyles } from '@material-ui/core/';
import { useDispatch } from 'react-redux';
import BaseButton from '../components/Buttons/BaseButton';
import PasswordInput from '../components/PasswordInput/PasswordInput';
import Layout from '../components/Layout/Layout';
import useText from '../commons/hooks/useText';
import RotatingLogo from '../components/RotatingLogo/RotatingLogo';
import { checkPassword } from '../pages/Background/actions';
import { ERROR_MSG, INVALID_PW } from '../commons/errors';
import dictionary from '../commons/dictionary';

/**
 * Component styles.
 * @type {StylesHook<Styles<{readonly spacing?: *, readonly palette?: *}, {}, string>>}
 */
const useStyles = makeStyles(({ palette, spacing }) => ({
  link: {
    color: palette.text.secondary,
    textAlign: 'center',
    marginTop: spacing(1),
  },
}));
/**
 * <UnlockWallet />
 * Base view which is shown when the user has some encrypted wallets but doesn't have the password in the password service.
 * @return {*}
 * @constructor
 */
const UnlockWallet = () => {
  const classes = useStyles();
  const dispatch = useDispatch();
  const { password, handleChange } = useText();
  const [error, setError] = useState(null);
  const [disabled, setDisabled] = useState(false);

  useEffect(() => {
    chrome.runtime.onMessage.addListener((request) => {
      if (request.msg === ERROR_MSG && request.error.code === INVALID_PW) {
        setDisabled(false);
        setError(request.error);
      }
    });
  }, []);

  /**
   * Check the password and set it in the password service.
   * @return {Promise<void>}
   */
  const unlock = async () => {
    setError(null);
    setDisabled(true);
    dispatch(checkPassword(password));
  };

  /**
   * Open the new browser tab with the import wallet form page.
   */
  const openFormTab = () => {
    chrome.tabs.create({ url: 'form.html' });
  };

  return (
    <Layout>
      <Box>
        <Box fontSize={24} color="text.primary">
          {dictionary.unlockWallet.title}
        </Box>
      </Box>

      <Box>
        <RotatingLogo />
      </Box>

      <Box>
        <PasswordInput
          value={password}
          handleChange={handleChange}
          hasError={!!error}
        />
        {!!error ? (
          <Box mt={1} fontSize="10px" color="text.error">
            {dictionary.unlockWallet.error}
          </Box>
        ) : (
          <Box mt={1} fontSize="10px" color="text.secondary">
            {dictionary.unlockWallet.info}{' '}
            <Link
              data-test-id={'import-wallet-link'}
              href="#"
              onClick={openFormTab}
              className={classes.link}
            >
              {dictionary.welcome.link.toLowerCase()}
            </Link>
          </Box>
        )}
      </Box>

      <Box
        display="flex"
        flexDirection="row"
        alignItems="center"
        justifyContent="center"
      >
        <BaseButton
          testID={'unlock-wallet-button'}
          disabled={disabled}
          handleClick={unlock}
          text={dictionary.unlockWallet.unlock}
          icon
        />
      </Box>
    </Layout>
  );
};

export default UnlockWallet;
