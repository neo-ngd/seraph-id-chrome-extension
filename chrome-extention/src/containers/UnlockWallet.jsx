import React, {useEffect, useState} from 'react';
import { Box } from '@material-ui/core/';
import {useDispatch } from 'react-redux';
import BaseButton from '../components/Buttons/BaseButton';
import PasswordInput from '../components/PasswordInput/PasswordInput';
import Layout from '../components/Layout/Layout';
import useText from '../commons/hooks/useText';
import RotatingLogo from '../components/RotatingLogo/RotatingLogo';
import {  checkPassword } from "../pages/Background/actions";
import {ERROR_MSG, INVALID_PW} from "../commons/errors";
import dictionary from "../commons/dictionary";

const UnlockWallet = () => {
  const dispatch = useDispatch();
  const { password, handleChange } = useText();
  const [ error, setError ] = useState(null);
  const [ disabled, setDisabled ] = useState(false);

    useEffect(() => {
        chrome.runtime.onMessage.addListener(request => {
            if (request.msg === ERROR_MSG && request.error.code === INVALID_PW) {
                setDisabled(false);
                setError(request.error);
            }
        });
    }, []);

  const unlock = async () => {
      setError(null);
      setDisabled(true);
      dispatch(checkPassword(password));
  };

  return (
    <Layout>
      <Box>
        <Box fontSize={24} color="text.primary">{dictionary.unlockWallet.title}</Box>
      </Box>

      <Box>
        <RotatingLogo />
      </Box>

      <Box>
        <PasswordInput value={password} handleChange={handleChange} hasError={!!error}/>
          {!!error ? <Box mt={1} fontSize="10px" color="text.error">{dictionary.unlockWallet.error}</Box> :
              <Box mt={1} fontSize="10px" color="text.secondary">{dictionary.unlockWallet.info}</Box>}
      </Box>

      <Box display="flex" flexDirection="row" alignItems="center" justifyContent="center" >
        <BaseButton
          testID={'unlock-wallet-button'}
          disabled={disabled}
          handleClick={unlock}
          text={dictionary.unlockWallet.unlock}
          icon
        />
      </Box>
    </Layout>
  );
}

export default UnlockWallet;
