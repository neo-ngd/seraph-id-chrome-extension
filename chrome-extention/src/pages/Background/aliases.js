import {
  ASK_CLAIM_ALIAS,
  CREATE_CLAIM_ALIAS,
  SET_PASSWORD_ALIAS,
  CHECK_PASSWORD_ALIAS,
  GET_PASSWORD_ALIAS,
  GET_PASSWORD_CS_ALIAS,
  SEND_ERROR_CS_ALIAS,
  SEND_ERROR_POPUP_ALIAS,
  IMPORT_WALLET_ALIAS,
  SHARE_ACTIVE_ACCOUNT,
} from './actionTypes';
import { createClaim, decrypt } from '../../commons/seraphSdkUtils';
import {
  setExportedWallet,
  setClaim,
  toggleDialog,
  setSession,
  getEncryptedPasswordToCS, setActiveAccount
} from './actions';
import { pwService } from "./pwService";
import {
  sendErrorToCS,
  invalidPwError,
  claimNotFoundError,
  unknownError,
  sendErrorToPopup,
  walletNotFoundError
} from "../../commons/errors";
import {
  ADD_CLAIM_ERROR_MSG,
  ADD_CLAIM_SUCCESS_MSG,
  DIALOG_TYPES,
  ENCRYPTED_PW_MSG,
  IMPORT_ERROR_MSG,
  IMPORT_SUCCESS_MSG,
  SHARE_ACCOUNT_MSG, SHARE_CLAIM_ERROR_MSG, SHARE_CLAIM_SUCCESS_MSG
} from "../../commons/constants";
import icon from '../../assets/icons/icon64.png';
import dictionary from "../../commons/dictionary";

const addClaimResultEvent = success => {
  chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
    chrome.tabs.sendMessage(tabs[0].id, {msg: success ? ADD_CLAIM_SUCCESS_MSG : ADD_CLAIM_ERROR_MSG });
  });
};

const shareClaimResultEvent = success => {
  chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
    chrome.tabs.sendMessage(tabs[0].id, {msg: success ? SHARE_CLAIM_SUCCESS_MSG : SHARE_CLAIM_ERROR_MSG });
  });
};

export const sendErrorToCSAlias = ({error}) =>
    () => sendErrorToCS(error);

export const sendErrorToPopupAlias = ({error}) =>
    () => sendErrorToPopup(error);

export const setPasswordAlias = ({password}) => (dispatch) =>  {
  pwService.password = password;
  return dispatch(setSession(true));
};

export const checkPasswordAlias = ({password}) => async (dispatch, getState) => {
  try {
    const { wallet } = getState();
    const decrypted = await decrypt(wallet, password);
    if (!!decrypted) {
      pwService.password = password;
      dispatch(setActiveAccount(decrypted.accounts[0].label));
      dispatch(getEncryptedPasswordToCS());
      dispatch(shareActiveAccountAlias());
      return dispatch(setSession(true));
    }
    return sendErrorToPopup(invalidPwError());
  } catch (error) {
    sendErrorToCS(unknownError(error))
  }
};

export const getEncryptedPasswordAlias = () => (dispatch, getState) => {
  const { wallet, activeAccount } = getState();
  chrome.runtime.sendMessage({msg: ENCRYPTED_PW_MSG, password: pwService.password, wallet, activeAccount});
};

export const getEncryptedPasswordCSAlias = () => (dispatch, getState) => {
  const { wallet, activeAccount } = getState();
  chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
    chrome.tabs.sendMessage(tabs[0].id, {msg: ENCRYPTED_PW_MSG, password: pwService.password, wallet, activeAccount});
  });
};

export const shareActiveAccountAlias = () => (dispatch, getState) => {
  const { activeAccount } = getState();
  chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
    chrome.tabs.sendMessage(tabs[0].id, {msg: SHARE_ACCOUNT_MSG, activeAccount });
  });
};

export const createClaimAlias = ({data, schemaName}) => async (dispatch, getState) => {
  try {
    const { wallet } = getState();
    const decryptedWallet = await decrypt(wallet, pwService.password);
    if (!!decryptedWallet) {
      const claim = createClaim(schemaName, data, decryptedWallet);
      decryptedWallet.addClaim(claim);
      const encryption = decryptedWallet.accounts.map(acc => acc.encrypt(pwService.password));
      await Promise.all(encryption);
      const exportedWalletJSON = JSON.stringify(decryptedWallet.export());
      chrome.notifications.create({
        type: 'basic',
        iconUrl: icon,
        title: '',
        message: dictionary.events.saveClaimSuccess});
      addClaimResultEvent(true);
      return dispatch(setExportedWallet(exportedWalletJSON));
    }
    addClaimResultEvent(false);
    return sendErrorToCS(walletNotFoundError())
  } catch (error) {
    addClaimResultEvent(false);
    sendErrorToCS(unknownError(error))
  }
};

export const askClaimAlias = ({schemaName, issuerDID, verifierName}) => async (dispatch, getState) => {
  try {
    const { wallet } = getState();
    const decryptedWallet = await decrypt(wallet, pwService.password);
    if (!!decryptedWallet) {
      const claims = decryptedWallet.getAllClaims(issuerDID);
      const claim = claims.find(claim => claim.schema === schemaName);
      if (!!claim) {
        dispatch(setClaim(claim));
        return dispatch(toggleDialog({open: true, context: DIALOG_TYPES.ASK_CLAIM, verifierName, schemaName}))
      }
      shareClaimResultEvent(false);
      return sendErrorToCS(claimNotFoundError())
    }
    shareClaimResultEvent(false);
    return sendErrorToCS(walletNotFoundError())
  } catch (error) {
    shareClaimResultEvent(false);
    sendErrorToCS(unknownError(error))
  }
};

export const importWalletAlias = ({wallet}) => async (dispatch) => {
  try {
    dispatch(setExportedWallet(wallet));
    chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
      chrome.tabs.sendMessage(tabs[0].id, {msg: IMPORT_SUCCESS_MSG});
    });
    chrome.notifications.create({
      type: 'basic',
      iconUrl: icon,
      title: '',
      message: dictionary.events.importWalletSuccess});
  } catch (error) {
    chrome.tabs.query({active: true, currentWindow: true}, (tabs) => {
      chrome.tabs.sendMessage(tabs[0].id, {msg: IMPORT_ERROR_MSG});
    });
  }
};

export default {
  [CREATE_CLAIM_ALIAS]: action => createClaimAlias(action),
  [ASK_CLAIM_ALIAS]: action => askClaimAlias(action),
  [SET_PASSWORD_ALIAS]: action => setPasswordAlias(action),
  [CHECK_PASSWORD_ALIAS]: action => checkPasswordAlias(action),
  [GET_PASSWORD_ALIAS]: () => getEncryptedPasswordAlias(),
  [GET_PASSWORD_CS_ALIAS]: () => getEncryptedPasswordCSAlias(),
  [SEND_ERROR_POPUP_ALIAS]: action => sendErrorToPopupAlias(action),
  [SEND_ERROR_CS_ALIAS]: action => sendErrorToCSAlias(action),
  [IMPORT_WALLET_ALIAS]: action => importWalletAlias(action),
  [SHARE_ACTIVE_ACCOUNT]: () => shareActiveAccountAlias(),
}
