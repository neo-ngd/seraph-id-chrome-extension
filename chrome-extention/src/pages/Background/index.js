import { createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import rootReducer from './reducers';
import { wrapStore, alias } from 'webext-redux';
import { composeWithDevTools } from 'remote-redux-devtools';
import { persistStore, persistReducer } from 'redux-persist';
import storageSession from 'redux-persist/lib/storage';
import aliases from './aliases'; // Use SessionStorage

const composeEnhancers = composeWithDevTools({
  realtime: true,
  hostname: 'localhost',
  port: 8000,
});
const persistConfig = {
  key: 'root',
  storage: storageSession,
  blacklist: ['password'],
};
const persistedReducer = persistReducer(persistConfig, rootReducer);
const middlewares = [alias(aliases), thunk];
export const store = createStore(persistedReducer, composeEnhancers(applyMiddleware(...middlewares)));
wrapStore(store);
export const persistor = persistStore(store);
