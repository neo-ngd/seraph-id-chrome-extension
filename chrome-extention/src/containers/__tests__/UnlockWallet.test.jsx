import React from 'react';
import { mount } from 'enzyme';
import UnlockWallet from "../UnlockWallet";
import configureStore from "redux-mock-store";
import {Provider} from "react-redux";
import chrome from "sinon-chrome";
import {CHECK_PASSWORD_ALIAS} from "../../pages/Background/actionTypes";

const middlewares = [];
const mockStore = configureStore(middlewares);

describe('<UnlockWallet />', () => {
    const store = mockStore({});

    beforeEach(() => {
        global.chrome = chrome;
    });

    afterEach(() => {
        store.clearActions();
    });

    it('should render component', () => {
        const wrapper = mount(
            <Provider store={store}>
                <UnlockWallet />
            </Provider>);
        expect(wrapper).toMatchSnapshot()
    });

    it('should unlock the wallet', () => {
        const wrapper = mount(
            <Provider store={store}>
                <UnlockWallet />
            </Provider>);
        wrapper.find({'data-test-id': 'unlock-wallet-button'}).first().simulate('click');
        const actions = store.getActions();
        expect(actions[0].type).toEqual(CHECK_PASSWORD_ALIAS);
    })
});
