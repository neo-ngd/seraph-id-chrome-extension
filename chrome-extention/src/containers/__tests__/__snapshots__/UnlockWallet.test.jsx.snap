// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnlockWallet /> should render component 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <UnlockWallet>
    <Layout>
      <Styled(MuiBox)
        display="flex"
        flex="0 0 100%"
        flexDirection="column"
        height="100%"
        justifyContent="space-between"
        maxWidth="100%"
        overflow="hidden"
      >
        <div
          className="MuiBox-root MuiBox-root-2"
        >
          <Styled(MuiBox)>
            <div
              className="MuiBox-root MuiBox-root-3"
            >
              <Styled(MuiBox)
                color="text.primary"
                fontSize={24}
              >
                <div
                  className="MuiBox-root MuiBox-root-4"
                >
                  Welcome back
                </div>
              </Styled(MuiBox)>
            </div>
          </Styled(MuiBox)>
          <Styled(MuiBox)>
            <div
              className="MuiBox-root MuiBox-root-5"
            >
              <RotationLogo
                maxWidth="230px"
              >
                <React3DRotationBox
                  minHeight="150px"
                  minWidth="auto"
                  perspective="200px"
                  rotateForce={20}
                >
                  <div
                    className="motion-box "
                    style={
                      Object {
                        "minHeight": "150px",
                        "minWidth": "auto",
                        "perspective": "200px",
                        "textAlign": "center",
                      }
                    }
                  >
                    <div
                      style={
                        Object {
                          "transform": "rotateX(0deg) rotateY(0deg)",
                          "transformStyle": "preserve-3d",
                        }
                      }
                    >
                      <img
                        alt="logo"
                        src="test-file-stub"
                        style={
                          Object {
                            "maxWidth": "230px",
                          }
                        }
                      />
                    </div>
                  </div>
                </React3DRotationBox>
              </RotationLogo>
            </div>
          </Styled(MuiBox)>
          <Styled(MuiBox)>
            <div
              className="MuiBox-root MuiBox-root-6"
            >
              <PasswordInput
                handleChange={[Function]}
                hasError={false}
                value=""
              >
                <WithStyles(ForwardRef(InputBase))
                  className="makeStyles-textField-8"
                  id="outlined-adornment-password"
                  onChange={[Function]}
                  placeholder="Password"
                  type="password"
                  value=""
                >
                  <ForwardRef(InputBase)
                    className="makeStyles-textField-8"
                    classes={
                      Object {
                        "adornedEnd": "MuiInputBase-adornedEnd",
                        "adornedStart": "MuiInputBase-adornedStart",
                        "disabled": "Mui-disabled",
                        "error": "Mui-error",
                        "focused": "Mui-focused",
                        "formControl": "MuiInputBase-formControl",
                        "fullWidth": "MuiInputBase-fullWidth",
                        "input": "MuiInputBase-input",
                        "inputAdornedEnd": "MuiInputBase-inputAdornedEnd",
                        "inputAdornedStart": "MuiInputBase-inputAdornedStart",
                        "inputHiddenLabel": "MuiInputBase-inputHiddenLabel",
                        "inputMarginDense": "MuiInputBase-inputMarginDense",
                        "inputMultiline": "MuiInputBase-inputMultiline",
                        "inputSelect": "MuiInputBase-inputSelect",
                        "inputTypeSearch": "MuiInputBase-inputTypeSearch",
                        "marginDense": "MuiInputBase-marginDense",
                        "multiline": "MuiInputBase-multiline",
                        "root": "MuiInputBase-root",
                      }
                    }
                    id="outlined-adornment-password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
                  >
                    <div
                      className="MuiInputBase-root makeStyles-textField-8"
                      onClick={[Function]}
                    >
                      <input
                        className="MuiInputBase-input"
                        id="outlined-adornment-password"
                        onAnimationStart={[Function]}
                        onBlur={[Function]}
                        onChange={[Function]}
                        onFocus={[Function]}
                        placeholder="Password"
                        type="password"
                        value=""
                      />
                    </div>
                  </ForwardRef(InputBase)>
                </WithStyles(ForwardRef(InputBase))>
              </PasswordInput>
              <Styled(MuiBox)
                color="text.secondary"
                fontSize="10px"
                mt={1}
              >
                <div
                  className="MuiBox-root MuiBox-root-29"
                >
                  To continue, please unlock your wallet
                </div>
              </Styled(MuiBox)>
            </div>
          </Styled(MuiBox)>
          <Styled(MuiBox)
            alignItems="center"
            display="flex"
            flexDirection="row"
            justifyContent="center"
          >
            <div
              className="MuiBox-root MuiBox-root-30"
            >
              <BaseButton
                disabled={false}
                fullWidth={false}
                handleClick={[Function]}
                icon={true}
                reject={false}
                small={false}
                testID="unlock-wallet-button"
                text="Unlock Wallet"
              >
                <WithStyles(ForwardRef(Button))
                  className="makeStyles-button-31"
                  color="inherit"
                  data-test-id="unlock-wallet-button"
                  disabled={false}
                  onClick={[Function]}
                  style={Object {}}
                  variant="contained"
                >
                  <ForwardRef(Button)
                    className="makeStyles-button-31"
                    classes={
                      Object {
                        "colorInherit": "MuiButton-colorInherit",
                        "contained": "MuiButton-contained",
                        "containedPrimary": "MuiButton-containedPrimary",
                        "containedSecondary": "MuiButton-containedSecondary",
                        "disabled": "Mui-disabled",
                        "focusVisible": "Mui-focusVisible",
                        "fullWidth": "MuiButton-fullWidth",
                        "label": "MuiButton-label",
                        "outlined": "MuiButton-outlined",
                        "outlinedPrimary": "MuiButton-outlinedPrimary",
                        "outlinedSecondary": "MuiButton-outlinedSecondary",
                        "root": "MuiButton-root",
                        "sizeLarge": "MuiButton-sizeLarge",
                        "sizeSmall": "MuiButton-sizeSmall",
                        "text": "MuiButton-text",
                        "textPrimary": "MuiButton-textPrimary",
                        "textSecondary": "MuiButton-textSecondary",
                      }
                    }
                    color="inherit"
                    data-test-id="unlock-wallet-button"
                    disabled={false}
                    onClick={[Function]}
                    style={Object {}}
                    variant="contained"
                  >
                    <WithStyles(ForwardRef(ButtonBase))
                      className="MuiButton-root makeStyles-button-31 MuiButton-contained MuiButton-colorInherit"
                      component="button"
                      data-test-id="unlock-wallet-button"
                      disabled={false}
                      focusRipple={true}
                      focusVisibleClassName="Mui-focusVisible"
                      onClick={[Function]}
                      style={Object {}}
                      type="button"
                    >
                      <ForwardRef(ButtonBase)
                        className="MuiButton-root makeStyles-button-31 MuiButton-contained MuiButton-colorInherit"
                        classes={
                          Object {
                            "disabled": "Mui-disabled",
                            "focusVisible": "Mui-focusVisible",
                            "root": "MuiButtonBase-root",
                          }
                        }
                        component="button"
                        data-test-id="unlock-wallet-button"
                        disabled={false}
                        focusRipple={true}
                        focusVisibleClassName="Mui-focusVisible"
                        onClick={[Function]}
                        style={Object {}}
                        type="button"
                      >
                        <button
                          className="MuiButtonBase-root MuiButton-root makeStyles-button-31 MuiButton-contained MuiButton-colorInherit"
                          data-test-id="unlock-wallet-button"
                          disabled={false}
                          onBlur={[Function]}
                          onClick={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          style={Object {}}
                          tabIndex={0}
                          type="button"
                        >
                          <span
                            className="MuiButton-label"
                          >
                            Unlock Wallet
                            <ForwardRef
                              className="makeStyles-arrowRight-33"
                              style={
                                Object {
                                  "opacity": 1,
                                }
                              }
                            >
                              <WithStyles(ForwardRef(SvgIcon))
                                className="makeStyles-arrowRight-33"
                                style={
                                  Object {
                                    "opacity": 1,
                                  }
                                }
                              >
                                <ForwardRef(SvgIcon)
                                  className="makeStyles-arrowRight-33"
                                  classes={
                                    Object {
                                      "colorAction": "MuiSvgIcon-colorAction",
                                      "colorDisabled": "MuiSvgIcon-colorDisabled",
                                      "colorError": "MuiSvgIcon-colorError",
                                      "colorPrimary": "MuiSvgIcon-colorPrimary",
                                      "colorSecondary": "MuiSvgIcon-colorSecondary",
                                      "fontSizeInherit": "MuiSvgIcon-fontSizeInherit",
                                      "fontSizeLarge": "MuiSvgIcon-fontSizeLarge",
                                      "fontSizeSmall": "MuiSvgIcon-fontSizeSmall",
                                      "root": "MuiSvgIcon-root",
                                    }
                                  }
                                  style={
                                    Object {
                                      "opacity": 1,
                                    }
                                  }
                                >
                                  <svg
                                    aria-hidden="true"
                                    className="MuiSvgIcon-root makeStyles-arrowRight-33"
                                    focusable="false"
                                    role="presentation"
                                    style={
                                      Object {
                                        "opacity": 1,
                                      }
                                    }
                                    viewBox="0 0 24 24"
                                  >
                                    <path
                                      d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
                                    />
                                  </svg>
                                </ForwardRef(SvgIcon)>
                              </WithStyles(ForwardRef(SvgIcon))>
                            </ForwardRef>
                          </span>
                          <NoSsr>
                            <WithStyles(undefined)
                              center={false}
                            >
                              <ForwardRef(TouchRipple)
                                center={false}
                                classes={
                                  Object {
                                    "child": "MuiTouchRipple-child",
                                    "childLeaving": "MuiTouchRipple-childLeaving",
                                    "childPulsate": "MuiTouchRipple-childPulsate",
                                    "ripple": "MuiTouchRipple-ripple",
                                    "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                    "rippleVisible": "MuiTouchRipple-rippleVisible",
                                    "root": "MuiTouchRipple-root",
                                  }
                                }
                              >
                                <span
                                  className="MuiTouchRipple-root"
                                >
                                  <TransitionGroup
                                    childFactory={[Function]}
                                    component={null}
                                    exit={true}
                                  />
                                </span>
                              </ForwardRef(TouchRipple)>
                            </WithStyles(undefined)>
                          </NoSsr>
                        </button>
                      </ForwardRef(ButtonBase)>
                    </WithStyles(ForwardRef(ButtonBase))>
                  </ForwardRef(Button)>
                </WithStyles(ForwardRef(Button))>
              </BaseButton>
            </div>
          </Styled(MuiBox)>
        </div>
      </Styled(MuiBox)>
    </Layout>
  </UnlockWallet>
</Provider>
`;
